### PR TITLE
Simplify setup.py install_requires and extras_require lists

### LINKED ISSUE #4860: Remove `extras_require`s that are redundant with policyengine-core

### PR DESCRIPTION
**Pull request by martinholmer**
Removes unneeded packages from `setup.py` install_requires list.  The removed packages are required by the `policyengine-core` package, so their inclusion here is redundant.

And also fixes #4860.

### CODE REVIEW

**Review by MaxGhenis — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -35,19 +34,8 @@
...
     install_requires=[
-        "click==8.1.3",
-        "h5py",
-        "microdf_python",
-        "pandas",
-        "pathlib",
         "policyengine-core",
-        "pytest",
-        "pytest-dependency",
         "pyyaml",
```
Comment (MaxGhenis) [inline setup.py:38]: we can remove this too - while it's not in -core, it's not imported anywhere

File: setup.py [hunk 2]
```diff
@@ -35,19 +34,8 @@
...
-        "requests",
-        "synthimpute",
-        "tables==3.9.2",
-        "tabulate",
         "tqdm",
     ],
     extras_require={
```
Comment (MaxGhenis) [inline setup.py:41]: for posterity, i filed #4860 to remove redundant ones here as well. no need to put in this pr though.

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: changelog_entry.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
+  changes:
+    fixed:
+    - Inclusion of unneeded packages in setup.py install_requires list.
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
      - Inclusion of unneeded packages in setup.py install_requires list.
```

File: setup.py (hunk 1 above)
Comment (MaxGhenis) [inline setup.py:38]:
```suggestion
```

File: setup.py (hunk 2 above)
Comment (martinholmer) [reply setup.py:41]: Making extras list changes in this PR.

File: setup.py (hunk 1 above)
Comment (martinholmer) [reply setup.py:38]: OK

File: setup.py [hunk 4]
```diff
@@ -35,43 +34,17 @@
...
-        "click==8.1.3",
-        "h5py",
-        "microdf_python",
-        "pandas",
-        "pathlib",
         "policyengine-core",
-        "pytest",
-        "pytest-dependency",
+        "microdf-python",
         "pyyaml",
```
Comment (MaxGhenis) [inline setup.py:39]:
```suggestion
```

Comment (martinholmer) [reply setup.py]: `pyyaml` is apparently never used by policyengine-us.

**Pull request merged**